sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -92,15 +92,50 @@ private[sql] trait LookupCatalog extends Logging {
    * Catalog name takes precedence over namespaces.
    */
   object CatalogAndNamespace {
-    def unapply(parts: Seq[String]): Some[(CatalogPlugin, Option[Seq[String]])] = parts match {
-      case Seq(catalogName, tail @ _*) =>
+    def unapply(nameParts: Seq[String]): Some[(CatalogPlugin, Seq[String])] = {
+      assert(nameParts.nonEmpty)
+      nameParts match {
+        case Seq(catalogName, tail @ _*) =>
+          try {
+            Some(
+              (catalogManager.catalog(catalogName), tail))
+          } catch {
+            case _: CatalogNotFoundException =>
+              Some((currentCatalog, nameParts))
+          }
+      }
+    }
+  }
+
+  /**
+   * Extract catalog and table identifier from a multi-part identifier with the current catalog if
+   * needed. Table identifier takes precedence over catalog name.
+   */
+  object CatalogAndTableIdentifier {
```

Comment:
It's not only for table, but also view, function as well. Shall we name it `CatalogAndIdentifier`? There is already a `CatalogAndIdentifier` and we need to unify them here.

Reply:
Since the existing `CatalogAndIdentifier` was a `private def`, I just renamed it to `CatalogAndMultipartIdentifier`. As discussed in a different thread, I will do a follow up to see we can replace `CatalogObjectIdentifier` with `CatalogAndIdentifier`